### PR TITLE
Fix config rexport

### PR DESCRIPTION
**Pull request by Mallets**
Rexport some config type under unstable_config feature.
Some type was erroneously removed by https://github.com/eclipse-zenoh/zenoh/pull/1419.

### CODE REVIEW

File: zenoh/src/lib.rs [hunk 1]
```diff
@@ -351,7 +351,11 @@ pub mod time {
 
 /// Configuration to pass to [`open`] and [`scout`] functions and associated constants
 pub mod config {
-    pub use zenoh_config::{WhatAmI, WhatAmIMatcher};
+    pub use zenoh_config::{EndPoint, Locator, WhatAmI, WhatAmIMatcher, ZenohId};
```
Comment (wyfo) [inline zenoh/src/lib.rs:354]: why putting back `EndPoint` and `Locator`?

Comment (fuzzypixelz) [reply zenoh/src/lib.rs:354]: They can't really be used from the stable API, having them there is superfluous I would say

Although users will write Endpoints in their configuration and they would benefit from the docs, it's still odd having a public item that once constructed can't be used in any way.

File: zenoh/src/lib.rs [hunk 2]
```diff
@@ -351,7 +351,11 @@ pub mod time {
...
+    #[zenoh_macros::unstable_config]
+    pub use zenoh_config::{
+        ModeDependentValue, PermissionsConf, PluginsConfig, SecretValue, ValidatedMap,
```
Comment (fuzzypixelz) [inline zenoh/src/lib.rs:357]: How was this chosen? It seems like an odd selection from what's available.

File: zenoh/src/lib.rs (hunk 1 above)
Comment (Mallets) [reply zenoh/src/lib.rs:354]: `Locator` is part of the `Hello` and scouting API. And they are stable.

File: zenoh/src/lib.rs (hunk 2 above)
Comment (Mallets) [reply zenoh/src/lib.rs:357]: I've simply reverted the changes.

**Pull request merged**